### DISCUSSION TITLE
Does nothing...

### DISCUSSION THREAD

**Discussion by Anyeos** [kex0/batch-face-swap#75] [General]
It only says it detect faces but does absolutely nothing. No one file is modified, no file is created, nothing happens. There are no error only progress about face detection and nothing more.

Can someone help me understand how it works? I want to use it only with one face or something simple. I don't need to do a massive face swap for a lot of images, only one image at time and see the results.

**Comment by kex0** [comment]
Check out extension called adetailer or roop

**Reply by mwilsongit** [reply]
I personally did the manual install after first fail but I had error message. My error was coming from openCV and there is a comment about it in the issues about just changing the settings to the fastest. But more on your problem... did you check the img to img output folder. if you have the extension visible in webUI and no error check the console to see if it detects a face. if yes and no error I think your problem isnt very big. Bon courage

**Reply by Anyeos** [reply]
Hi, thanks for answers. I have no time now for test but I remember that eventually using fastest settings appears to do something but there are no other visible results. No error and nothing appears to be happening. May I don't figured out how it should be used.
Anyway I don't end to understand where to put the face to replace with.
I will do more test when I had the time but now I only thank you for the answer. Have a nice day.

**Reply by rubberbaron** [reply]
You don't supply a face to swap with. batch-face-swap uses Stable Diffusion inpainting to replace the faces.

You can get the exact same effect for a single image by simply inpainting the face you want to change in img2img. The purpose of batch-face-swap is to automate inpainting the faces so you don't have to manually draw the mask for every face. If you only want to process one face at a time and you can easily draw the mask yourself, there's no reason to use batch-face-swap. If you want to replace with some specific face, not use a prompt to specify how Stable Diffusion should synthesize a new one, then batch-face-swap is not the right extension.
